auslib/web/admin/swagger/api.yaml

```diff
@@ -0,0 +1,423 @@
+    description: "Invalid Request's Response JSON Object."
+    type: object
+    # currently we're using detail and status only in the UI
+    # Therefore rest of fields aren't always required.
+    required:
+      - detail
+      - status
+    properties:
+      detail:
+        description: An human readable explanation specific to this occurrence of the problem.
+        type: string
+        example: "field_name is a required property"
+      status:
+        description: The HTTP status code generated by the origin server for this occurrence of the problem.
+        type: integer
+        format: int32
+        enum: [400]
+        example: 400
+      title:
+        description: A short, human-readable summary of the problem type. It SHOULD NOT change from occurrence to occurrence of the problem, except for purposes of localisation.
+        type: string
+        example: "Bad Request"
+      type:
+        description: "An absolute URI that identifies the problem type.  When dereferenced, it SHOULD provide human-readable documentation for the problem type (e.g., using HTML).When this member is not present its value is assumed to be 'about:blank'"
+        type: string
+        example: "about:blank"
+
+  CSRFModel:
+    title: CSRF Token
+    description: CSRF Token in POST
+    type: object
+    required:
+      - csrf_token
+    properties:
+      csrf_token:
+        type: string
+        example: "1491342563##c4e6fef0b978e6c89af9ff1015e67b9ca7c45d14"
+
+  RulesBaseGET:
+    title: GET Operations on Rules Base
+    description: "Sample description for /rules GET"
+    allOf:
+      - $ref: "#/definitions/RulesBase"
+    required:
+      - alias
+      - backgroundRate
+      - buildID
+      - buildTarget
+      - channel
+      - comment
+      - data_version
+      - distVersion
+      - distribution
+      - fallbackMapping
+      - headerArchitecture
+      - locale
+      - mapping
+      - osVersion
+      - priority
+      - product
+      - rule_id
+      - systemCapabilities
+      - update_type
+      - version
+
+  RulesBase:
+    title: Rules object definition
+    description: Balrog's Rule Attributes
+    type: object
+    required:
+      - update_type # Required during both POSTing and GET(fetch) Operations
+
+    # Cannot have additionalProperties as false for strict validation ,else it disallows csrf_token or any other model's properties to be added
+    #additionalProperties: false
+
+    properties:
+      rule_id:
+        description: The id of the rule. This id is necessary to make changes to the rule through the REST API.
+        type: integer
+        format: int32
+        readOnly: true
+        example: 10
+
+      priority:
+        description: The priority of the rule, relative to other rules. If multiple rules match an incoming request based on the Matchable columns, the rule with the highest priority is chosen.
+        type: ["integer", "string", "null"]
```

Comment:
Alright once this is merged , then I should file a bug so that I can write the pointers to the assignee where to make code changes both in front and back end.